apps/obsidian/src/components/CreateNodeModal.tsx

```diff
@@ -0,0 +1,143 @@
+import { App, Modal, Notice } from "obsidian";
+import { DiscourseNode } from "~/types";
+import type DiscourseGraphPlugin from "~/index";
+
+interface CreateNodeModalProps {
+  nodeTypes: DiscourseNode[];
+  plugin: DiscourseGraphPlugin;
+  onNodeCreate: (nodeType: DiscourseNode, title: string) => Promise<void>;
+}
+
+export class CreateNodeModal extends Modal {
+  private nodeTypes: DiscourseNode[];
+  private plugin: DiscourseGraphPlugin;
+  private onNodeCreate: (
+    nodeType: DiscourseNode,
+    title: string,
+  ) => Promise<void>;
+  private selectedNodeType: DiscourseNode | null = null;
+  private titleInput: HTMLInputElement | null = null;
+  private confirmButton: HTMLButtonElement | null = null;
+
+  constructor(app: App, props: CreateNodeModalProps) {
+    super(app);
+    this.nodeTypes = props.nodeTypes;
+    this.plugin = props.plugin;
+    this.onNodeCreate = props.onNodeCreate;
+  }
+
+  onOpen() {
+    const { contentEl } = this;
+    contentEl.empty();
+
+    contentEl.createEl("h2", { text: "Create node" });
+
+    const titleContainer = contentEl.createDiv({ cls: "setting-item" });
+    titleContainer.createEl("div", { text: "Title", cls: "setting-item-name" });
+    const titleInputContainer = titleContainer.createDiv({
+      cls: "setting-item-control",
+    });
+
+    this.titleInput = titleInputContainer.createEl("input", {
```

Comment:
This felt quite small, especially considering how long some of the nodes in `megacoglab` and `akamatsulab` are.  Likely we'll more room here so the user can see as they type out node names.

Can be deferred.